### PR TITLE
Add option for overriding the default compression algorithm when publishing images

### PR DESCRIPTION
**Pull request by cneira**
Add option for overriding the default compression algorithm when publishing images
Changes done as suggested by stgraber

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -2414,12 +2414,11 @@ func (c *Client) AsyncWaitMeta(resp *Response) (*shared.Jmap, error) {
...
 }
 
-func (c *Client) ImageFromContainer(cname string, public bool, aliases []string, properties map[string]string) (string, error) {
+func (c *Client) ImageFromContainer(cname string, public bool, aliases []string, properties map[string]string, compression_algorithm string) (string, error) {
 	if c.Remote.Public {
 		return "", fmt.Errorf("This function isn't supported by public remotes.")
 	}
-
-	source := shared.Jmap{"type": "container", "name": cname}
+	source := shared.Jmap{"type": "container", "name": cname, "compression_algorithm": compression_algorithm}
```
Comment (stgraber) [inline client.go]: compression_algorithm should only be added to the source map if it's non-empty

File: doc/api_extensions.md [hunk 2]
```diff
@@ -74,3 +74,6 @@ flag to each XHR Request).
 Some browsers like Firefox and Safari can't accept server response without
 `Access-Control-Allow-Credentials: true` header. To ensure that the server will
 return a response with that header, set `core.https_allowed_credentials=true`.
+
+##image compression
+Add support to specify compression algorithm when publishing an image
```
Comment (stgraber) [inline doc/api_extensions.md]: This should mention that this is the compression_algorithm property on POST /1.0/images

**Comment by stgraber** [comment]
Left a couple more comments. You're also still missing the change to doc/rest-api.md

**Comment by stgraber** [comment]
Going to take this into my branch, do the few needed fixes and send it for review.

**Comment by cneira** [comment]
Thank you very much Stéphane. I'm short on time to fix the rest myself :).
I'll check tomorrow which other issues could I start working to get my feet
wet on lxd.

Bests

**Comment by stgraber** [comment]
Closing in favor of #2348

**Pull request closed without merging**